pages/api/auth/[...nextauth].ts

```diff
@@ -1,70 +1,135 @@
-    },
-    async session({ session, token }) {
-      // console.log(`Session`, session, user, token);
-      return {
-        expires: session.expires,
-        user: {
-          ...session.user,
-          login: token.login,
-          accessToken: token?.access_token,
-        },
-      };
-    },
-  },
-});
+import { NextApiRequest, NextApiResponse } from "next";
+import NextAuth, { Account, Profile } from "next-auth";
+import { getToken } from "next-auth/jwt";
+import getConfig from "next/config";
+
+import models from "db/models";
+
+import { DAY_IN_SECONDS } from "helpers/constants";
+import { caseInsensitiveEqual } from "helpers/db/conditionals";
+import { toLower } from "helpers/string";
+
+import { Logger } from "services/logging";
+
+import { EthereumProvider, GHProvider } from "server/auth/providers";
+
+const {
+  publicRuntimeConfig,
+  serverRuntimeConfig: {
+    auth: {
+      secret
+    }
+  }
+} = getConfig();
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+  const currentToken = await getToken({ req, secret: secret });
+
+  return NextAuth(req, res, {
+    providers: [
+      EthereumProvider,
+      GHProvider
+    ],
+    pages: {
+      signIn: "/auth/signin"
+    },
+    session:{
+      strategy: "jwt",
+      maxAge: 30 * DAY_IN_SECONDS
+    },
+    callbacks: {
+      async signIn({ profile, account}) {
+        try {
+          const provider = account?.provider;
+
+          if (provider === "github") {
+            if (!profile?.login) return "/?authError=Profile not found";
+
+            return true;
+          } else if (provider === "credentials") {
+            const accountAddress = account?.providerAccountId;
+
+            if (accountAddress) {
+              await models.user.findOrCreate({
+                where: {
+                  address: caseInsensitiveEqual("address", accountAddress.toLowerCase())
+                },
+                defaults: {
+                  address: accountAddress
+                }
+              });
+  
+              return true;
+            }
+          }
+        } catch(error) {
+          Logger.error(error, "SignIn callback: ");
+        }
+
+        return false;
+      },
+      async jwt({ token, profile, account }) {
+        try {
+          const provider = account?.provider;
+          
+          if (provider === "github") {
+            const { name, login } = profile;
+            const { provider, access_token } = account;
+
+            return { 
+              ...token, 
+              profile: {
+                name,
+                login,
+              },
+              account: {
+                provider,
+                access_token
+              }, 
+              address: currentToken?.address,
+              role: currentToken?.role,
+            };
+          } else if (provider === "credentials") {
+            const address = token?.sub;
+
+            let role = "user";

```

Comment:
make role be an array, that way we can have `[user, governor, admin]` (because this might be true, and because we can fine-tune the requests based on jwt role via .includes()), also use a constant `ROLE_USER = "user"` so we can edit it easily if needed.

note: a governor is the user who deployed a network, not the user who is mentioned on the adminWallet; We should call this person an `admin` -- which can also be a governor, but that we check to see if this address matches any of the creatorAddress on our networks table